src/pages/_middleware.ts

```diff
@@ -1,17 +1,16 @@
 import { NextFetchEvent, NextRequest, NextResponse } from "next/server";
 
 export async function middleware(req: NextRequest, ev: NextFetchEvent) {
-  if (
-    req.nextUrl.pathname.startsWith("/api/") ||
-    req.nextUrl.pathname === "/"
-  ) {
+  if (req.nextUrl.pathname.startsWith("/api/") || req.nextUrl.pathname === "/") {
     return;
   }
   const slug = req.nextUrl.pathname.split("/").pop();
 
-  const data = await (
-    await fetch(`${req.nextUrl.origin}/api/get-url/${slug}`)
-  ).json();
+  const slugFetch = await fetch(`${req.nextUrl.origin}/api/get-url/${slug}`)
+  if (slugFetch.status === 404) {
+    return NextResponse.redirect(req.nextUrl.origin)
```

Comment:
I don't think you have prettier installed and it's fuckin w/ my formatting lol

Reply:
Lol fixing that now